app/src/main/java/com/cornellappdev/score/util/DateUtil.kt

```
@@ -66,6 +68,21 @@ fun parseDateTimeOrNull(date: String, time: String): LocalDateTime? {
+/**
+ * Parses an ISO-8601 timestamp: yyyy-MM-dd'T'HH:mm:ss'Z' into a LocalDateTime object.
+ *
+ * @param strDate the date string to parse, in the format "yyyy-MM-dd'T'HH:mm:ss'Z'"
+ * @return a LocalDateTime object if parsing succeeds, or null if the format is invalid
+ */
+fun parseIsoDateToLocalDateOrNull(strDate: String): LocalDate? {
+    return try {
+        Instant.parse(strDate)
+            .atZone(ZoneId.systemDefault())
+            .toLocalDate()
+    } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)